eox_hooks/actions.py

```diff
@@ -0,0 +1,70 @@
+import requests
+from django.conf import settings
+
+from eox_hooks.serializers import CertificateSerializer,UserSerializer
+from eox_hooks.utils import _get_action_settings, _get_course, flatten
+from lms.djangoapps.course_api.serializers import CourseDetailSerializer
+
+
+def send_request(**kwargs):
+    """
+    Send data to a webhook url specified in the microsite eox-hooks settings.
+    """
+    action_settings = _get_action_settings("send_request")
+    webhook_url = action_settings.get("url", "")
+    fields = action_settings.get("fields", {})
+    extra_fields = action_settings.get("extra_fields", {})
+
+    data = _get_request_fields(fields, extra_fields, **kwargs)
+
+    if action_settings.get('send_extended_certificate_info', False):
+        certificate = kwargs.get('certificate', {})
+        extended_data = get_extended_certificate_data(certificate)
+        data.update(extended_data)
+
+    result = requests.post(webhook_url, flatten(data))
+    return result.status_code == 200
+
+
+def get_extended_certificate_data(certificate):
+    """
+    Add to the request all the information from a certificate, 
+    the user and the course associated to it.
+    """
+    course = _get_course(certificate.course_id)
+    user_serializer = UserSerializer(certificate.user)
+    certificate_serializer = CertificateSerializer(certificate)
+    course_serializer = CourseDetailSerializer(course)
+
+    extended_data = {
+        'certificate': certificate_serializer.data,
+        'user': user_serializer.data,
+        #'course': course_serializer.data
+    }
+
+    return extended_data
+
+
+def _get_request_fields(fields, extra_fields, **kwargs):
+    """
+    Retrieve a dictionary with all the fields specified in the configs.
+    """
+    data = {}
+
+    for name,field in fields.items():
+        try:
+            attributes = field.split(".")
+            obj = kwargs.get(attributes.pop(0))
+
+            for attr in attributes:
+                obj = getattr(obj, attr, None)
+        except:
+            obj = None
+
+        data[name] = str(obj)
+
+    # Include extra-fields
+    for name, field in extra_fields.items():
+        data[name] = field
```

Comment:
Are those fields static i.e. we are just using the information from the settings to fill them?

Reply:
Yes, I added this in case someone wanted to add specific values directly from the settings. What do you think?
Also, I'm aware that mi PR description needs improvement, I'm working on that.